packages/blueprints-integration/src/abPlayback.ts

```diff
@@ -0,0 +1,70 @@
+import type { TimelineObjectAbSessionInfo } from '@sofie-automation/shared-lib/dist/core/model/Timeline'
+import type { ICommonContext } from './context'
+import type { OnGenerateTimelineObj, TSR } from './timeline'
+
+export interface PieceAbSessionInfo extends TimelineObjectAbSessionInfo {
+	/**
+	 * If true, this assignment will be given low priority
+	 */
+	optional?: boolean
+}
+
+/**
+ * A token to identify sessions that need an assignment, but won't be able to provide a unique id.
+ * Instead, when this token is encountered, the id of the pieceInstance will be used instead of the name of the session.
+ * Note: This doesnt play nice with transitions, so if they are required, then auto cannot be used
+ */
+export const AB_MEDIA_PLAYER_AUTO = '__auto__'
+
+/**
+ * A rule describing how to update the `layer` of a TimelineObject for AB playback
+ */
+export interface ABTimelineLayerChangeRule {
+	/** What AB pools can this rule be used for */
+	acceptedPoolNames: string[]
+	/** A function to generate the new layer name for a chosen playerId */
+	newLayerName: (playerId: number) => string
+	/** Whether this rule can be used for lookaheadObjects */
+	allowsLookahead: boolean
+}
+
+/**
+ * A set of rules describing how to update the `layer` of a TimelineObject for AB playback
+ */
+export type ABTimelineLayerChangeRules = {
+	[fromLayer: string]: ABTimelineLayerChangeRule | undefined
+}
+
+/**
+ * Configuration of the AB playback functionality provided by Sofie
+ */
+export interface ABResolverConfiguration {
+	/** Options for the resolver */
+	resolverOptions: ABResolverOptions
+	/** The AB pools that should be processed, and the playerIds in each pool */
+	pools: Record<string, number[]>
```

Comment:
Maybe this should be an array of objects, for forwards-compatibility?
```suggestion
	pools: Record<string, Array<{ playerId: number }>>
```